implementations/micrometer-registry-new-relic/src/main/java/io/micrometer/newrelic/NewRelicMeterRegistry.java

```diff
@@ -15,82 +15,79 @@
+import io.micrometer.core.instrument.util.NamedThreadFactory;
+import io.micrometer.core.instrument.util.TimeUtils;
 
 /**
- * Publishes metrics to New Relic Insights.
+ * Publishes metrics to New Relic Insights based on client provider selected (Http or Java Agent). 
+ * Defaults to the HTTP/REST client provider.
  *
  * @author Jon Schneider
  * @author Johnny Lim
- * @since 1.0.0
+ * @author Neil Powell
  */
 public class NewRelicMeterRegistry extends StepMeterRegistry {
+
     private static final ThreadFactory DEFAULT_THREAD_FACTORY = new NamedThreadFactory("new-relic-metrics-publisher");
     private final NewRelicConfig config;
-    private final HttpSender httpClient;
+    private NewRelicClientProvider clientProvider;
     private final Logger logger = LoggerFactory.getLogger(NewRelicMeterRegistry.class);
 
     /**
      * @param config Configuration options for the registry that are describable as properties.
      * @param clock  The clock to use for timings.
      */
-    @SuppressWarnings("deprecation")
     public NewRelicMeterRegistry(NewRelicConfig config, Clock clock) {
-        this(config, clock, DEFAULT_THREAD_FACTORY,
-                new HttpUrlConnectionSender(config.connectTimeout(), config.readTimeout()));
+        //default to the HTTP/REST client
+        this(config, new NewRelicHttpClientProviderImpl(config), clock);
+    }
+    
+    /**
+     * @param config Configuration options for the registry that are describable as properties.
+     * @param clientProvider Provider of the HTTP or Agent-based client that publishes metrics to New Relic
+     * @param clock  The clock to use for timings.
+     */
+    public NewRelicMeterRegistry(NewRelicConfig config, NewRelicClientProvider clientProvider, Clock clock) {
+        this(config, clientProvider, new NewRelicNamingConvention(), clock, DEFAULT_THREAD_FACTORY);
     }
 
     /**
-     * @param config        Configuration options for the registry that are describable as properties.
-     * @param clock         The clock to use for timings.
+     * @param config Configuration options for the registry that are describable as properties.
+     * @param clientProvider Provider of the HTTP or Agent-based client that publishes metrics to New Relic
+     * @param namingConvention Naming convention to apply before metric publishing
+     * @param clock  The clock to use for timings.
      * @param threadFactory The thread factory to use to create the publishing thread.
      * @deprecated Use {@link #builder(NewRelicConfig)} instead.
      */
-    @Deprecated
-    public NewRelicMeterRegistry(NewRelicConfig config, Clock clock, ThreadFactory threadFactory) {
-        this(config, clock, threadFactory, new HttpUrlConnectionSender(config.connectTimeout(), config.readTimeout()));
-    }
-
-    // VisibleForTesting
-    NewRelicMeterRegistry(NewRelicConfig config, Clock clock, ThreadFactory threadFactory, HttpSender httpClient) {
+    @Deprecated  // VisibleForTesting
```

Comment:
No need to mark this deprecated if it is package-private. We marked it before because it was public API. It's been deprecated since 1.1.0, so it is probably fair to remove it from public API at this point.

Reply:
Of course.